syntax/format_test.go

```diff
@@ -0,0 +1,77 @@
+package syntax
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFmtPrettyForDict(t *testing.T) { //nolint:dupl
+	t.Parallel()
+	simpleDict, err := EvaluateExpr(".", `{'a':1, 'b':2, 'c':3}`)
+	assert.Nil(t, err)
+	str := FormatString(simpleDict, 0)
+	assert.Equal(t, true, contains(str, "\n%sa: 1", 1))
```

Comment:
Nit: replace all `assert.Equal(t, true` with `assert.True(t`

Reply:
Actually, the `contains` checks don't seem like a good fit for what you're trying to test. I'd expect to see something of the form:

```
str := FormatString(simpleDict, 0)
assert.Equal(t, `{
  a: 1,
  b: 2,
  c: 3
}`, str)
```
Does that not work for some reason?

Reply:
This can't work as the evaluation result is not order guaranteed. `EvaluateExpr(".", '(a:1, b:2, c:3)')` can't output 
```
{
  a: 1, 
  b: 2, 
  c: 3
}
```
always.

Reply:
That seems like a bug... Marcelo's guidance has generally been avoid nondeterminism where possible, since it makes everything more complicated.

In this case, it would be quite desirable for a pretty printer to produce deterministic output, especially deterministically sorted output. The items in tuples, dicts and sets should all be ordered in the pretty output, a) so we can test more easily, and b) because it's easier for a user to read.

Is that possible?